pkg/oauth/registry/selfoauthclientauthorization/etcd/etcd.go

```diff
@@ -32,7 +32,7 @@ func NewREST(optsGetter restoptions.Getter) (*REST, error) {
 
 	store := &registry.Store{
 		NewFunc:     func() runtime.Object { return &api.SelfOAuthClientAuthorization{} },
-		NewListFunc: func() runtime.Object { return &api.SelfOAuthClientAuthorizationList{} },
+		NewListFunc: func() runtime.Object { return &api.OAuthClientAuthorizationList{} },
```

Comment:
I feel like there is some way to avoid having to use `OAuthClientAuthorizationList` here (i.e. use the `Self` version).